pkg/deployments/DEPLOYING.md

```diff
@@ -0,0 +1,135 @@
+# Deploying Balancer Contracts
+
+The deployments package aims to provide a robust mechanism to run and verify deployments on multiple networks. This document describes rationale, what guarantees it provides, and how to create and run deployments.
+
+## Overview
+
+Each deployment unit is called a 'task'. Tasks have a unique ID, which is often prefixed with the date in which they were first executed (to e.g. tell apart the deployment of Weighted Pools from April 2021, and the one in July 2022, which might feature widly different behavior).
+
+A task is made up of multiple components:
+
+- one or multiplle build information JSON files, located in the `build-info` directory.
+- an `input.ts` file, containing information on dependencies and deployment parameters (which might differ accross networks).
+- an `index.ts` file, with the instructions for the actual deployment.
+- the `output`, `abi` and `bytecode` directories, with JSON files containing the deployment addresses of the tasks' contracts in each network as well as additional information.
+- optional fork tests in the `test` directory.
+
+### Why `build-info`
+
+When Hardhat compiles a Solidity project, in addition to the traditional contract artifacts it also produces a special file in the `artifacts/build-info` directory, which includes all of the compilation inputs (full source code, compiler version and settings, etc.) and all of the compilation outputs (creation bytecode, ABI, AST, etc.). By storing this file in the repository and using it as the source of truth for deployments, we fully break the dependency on the source code and are able to modify it and update it after a deployment is created, while guaranteeing that all deployments of a given task will be identical (since they share the same build information).
+
+As part of our development process, we frequently alter the source code of contracts that have already been deployed, and keep no historical versions: the deployment data is fully stored in the build information file.
+
+## Creating Tasks
+
+A new task is created by placing a new directory named after the task ID in the `tasks`, and populating the `build-info` directory and writing appropiate `input.ts` and `index.ts` files. The `output`, `abi` and `bytecode` directories are automatically generated and populated.
+
+The build information should be generated on a clean commit, so that it can be verified by other parties. On the project directory, delete the `artifacts` directory and run `yarn hardhat compile` for a clean compilation. The file will be then be located at `artifacts/build-info`.
```

Comment:
Do we need to explain about editing package.json to remove the part that normally deletes build-info on compilation? Or make a compile-with-buildinfo?

Reply:
`yarn hardhat compile` will do the right thing. I don't want to go into too much detail tbh - this is not necessarily a step-by-step guide. And the reason why we delete the buldinfos is entirely accidental.